tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterPresenterSpecs.cs

```diff
@@ -130,49 +154,267 @@ protected static MetaDataCategory createMetaDataCategory<T>(string descriptiveNa
+
+      protected override void Context()
+      {
+         base.Context();
+         var dataFormat = A.Fake<IDataFormat>();
+         A.CallTo(() => dataFormat.Parameters).Returns(Enumerable.Empty<DataFormatParameter>().ToList());
+         _args = new FormatChangedEventArgs() { Format = dataFormat };
+      }
+
+      protected override void Because()
+      {
+         _importerDataPresenter.OnFormatChanged += Raise.With(_args);
+      }
+
+      [Observation]
+      public void invokes_column_mapping_presenter()
+      {
+         A.CallTo(() => _columnMappingPresenter.SetDataFormat(_args.Format)).MustHaveHappened();
+      }
+   }
+
+   public class When_tab_changed : concern_for_ImporterPresenter
+   {
+      protected TabChangedEventArgs _args;
+
+      protected override void Context()
+      {
+         base.Context();
+         _args = new TabChangedEventArgs() { TabData = new UnformattedData() };
+      }
+
+      protected override void Because()
+      {
+         _importerDataPresenter.OnTabChanged += Raise.With(_args);
+      }
+
+      [Observation]
+      public void invokes_column_mapping_presenter()
+      {
+         A.CallTo(() => _columnMappingPresenter.SetRawData(_args.TabData)).MustHaveHappened();
+      }
+   }
+
+   public class When_mapping_completed_with_loaded_data : concern_for_ImporterPresenter
+   {
+      protected override void Because()
+      {
+         base.Because();
+         A.CallTo(() => _dataSource.ValidateDataSource(A<IReadOnlyList<ColumnInfo>>.Ignored, A<IDimensionFactory>.Ignored)).Returns(true);
+         var sheets = new Cache<string, DataSheet>();
+         sheets.Add("sheet1", A.Fake<DataSheet>());
+         _importerDataPresenter.OnImportSheets += Raise.With(new ImportSheetsEventArgs() { Filter = "", DataSourceFile = _dataSourceFile, Sheets = sheets });
+         _columnMappingPresenter.OnMappingCompleted += Raise.With(new EventArgs());
+      }
+
+      [Observation]
+      public void invokes_column_mapping_presenter()
+      {
+         A.CallTo(() => _importerDataPresenter.onCompletedMapping()).MustHaveHappened();
+      }
+
+      [Observation]
+      public void shows_confirmation_view()
+      {
+         A.CallTo(() => _importerView.EnableConfirmationView()).MustHaveHappened();
+      }
+   }
+
+   public class When_mapping_completed_without_loaded_data : concern_for_ImporterPresenter
+   {
+      protected override void Because()
+      {
+         A.CallTo(() => _dataSource.ValidateDataSource(A<IReadOnlyList<ColumnInfo>>.Ignored, A<IDimensionFactory>.Ignored)).Returns(true);
+         _columnMappingPresenter.OnMappingCompleted += Raise.With(new EventArgs());
+      }
+
+      [Observation]
+      public void does_not_show_confirmation_view()
+      {
+         A.CallTo(() => _importerView.EnableConfirmationView()).MustNotHaveHappened();
+      }
+   }
+
+   public class When_mapping_completed_with_loaded_data_with_wrong_mapping : concern_for_ImporterPresenter
```

Comment:
@abdelr I don't think this tests does anything because there is NO action in because. All the because should be context IMO